### PR TITLE
test: improve ptk_shell tests

### PR DESCRIPTION
**Pull request by jnoortheen**
## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

### CODE REVIEW

File: tests/test_ptk_shell.py [hunk 1]
```diff
@@ -126,10 +126,17 @@ def test_remove_ansi_osc(raw_prompt, prompt, osc_tokens):
...
     inp, out, shell = ptk_shell
-    text = "hello"
-    inp.send_text(f"{text}\n")  # note: terminate with '\n'
-    result = shell.singleline()
+    inp.send_text(f"{line}\nexit\n")  # note: terminate with '\n'
+    shell.cmdloop()
+    out, err = capsys.readouterr()
     # todo: check rendered output using https://pyte.readthedocs.io/
-    assert result == text
+    assert out.strip() == exp or line
```
Comment (daniel-shimon) [inline tests/test_ptk_shell.py:142]: Again, what's `line` doing here?

Comment (jnoortheen) [reply tests/test_ptk_shell.py:142]: when exp is not given it is expected to be same as input. 
like when giving strings

Comment (daniel-shimon) [reply tests/test_ptk_shell.py:142]: So you need to put parentheses around the `or` expression

Comment (jnoortheen) [reply tests/test_ptk_shell.py]: Done!

**Review by gforsyth — APPROVED** [review]
This is very slick @jnoortheen -- I desperately wanted something like this back in 2015, glad to see it coming together now!

**Pull request merged**